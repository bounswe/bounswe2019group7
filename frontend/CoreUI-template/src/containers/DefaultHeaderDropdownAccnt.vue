<template>
  <AppHeaderDropdown right no-caret>
    <template slot="header">
      <img src="img/avatars/user.png" class="img-avatar" />
    </template>\
    <template slot="dropdown">
      <b-dropdown-header tag="div" class="text-center">
        <strong>Settings</strong>
      </b-dropdown-header>
      <b-dropdown-item :to="'/profilePage'">
        <i class="fa fa-user" /> Profile
      </b-dropdown-item>
      <b-dropdown-item :to="'/userSettings'">
        <i class="fa fa-wrench" /> Settings
      </b-dropdown-item>
      <b-dropdown-item :to="'/tradingAccount'" v-if="seen">
        <i class="fa fa-usd" /> Trading Account
      </b-dropdown-item>
      <b-dropdown-item>
        <i class="fa fa-usd" /> Payments
      </b-dropdown-item>
      <b-dropdown-divider />
      <b-dropdown-item v-on:click="logout">
        <i class="fa fa-lock" /> Logout
      </b-dropdown-item>
    </template>
  </AppHeaderDropdown>
</template>

<script>
import { HeaderDropdown as AppHeaderDropdown } from "@coreui/vue";
export default {
  name: "DefaultHeaderDropdownAccnt",
  components: {
    AppHeaderDropdown
  },
  data() {
    return {
      seen: false,
      item: []
    };
  },
  methods: {
    logout: function() {
      //TODO delete the cookie
      localStorage.removeItem("token");
      localStorage.removeItem("notification");
      localStorage.removeItem("email");
      localStorage.removeItem("password");
      document.location.href = ""; //mainpage -- login
    }
  },
  mounted() {
    this.$http
      .get("http://100.26.202.213:8080/user_profile/self_profile", {
        headers: {
          Authorization: localStorage.getItem("token")
        }
      })
      .then(
        response => {
          this.item = response.data;
          if (this.item.role == "BASIC_USER") {
            this.seen = false;
          } else {
            this.seen = true;
          }
        },
        error => {
          console.log("eerror");
        }
      );
  }
};
</script>
