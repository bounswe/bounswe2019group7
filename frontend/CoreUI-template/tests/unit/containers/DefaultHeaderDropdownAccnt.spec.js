import Vue from 'vue'
import BootstrapVue from 'bootstrap-vue'
import DefaultHeaderDropdownAccnt from '@/containers/DefaultHeaderDropdownAccnt'
import { shallowMount } from '@vue/test-utils';

Vue.use(BootstrapVue)

describe('DefaultHeaderDropdownAccnt.vue', () => {
  it('has a name', () => {
    expect(DefaultHeaderDropdownAccnt.name).toMatch('DefaultHeaderDropdownAccnt')
  })
  it('has a created hook', () => {
    expect(typeof DefaultHeaderDropdownAccnt.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof DefaultHeaderDropdownAccnt.data).toMatch('function')
    const defaultData = DefaultHeaderDropdownAccnt.data()
    expect(defaultData.itemsCount).toBe(42)
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(DefaultHeaderDropdownAccnt)
    expect(wrapper.element).toMatchSnapshot()
  })
})
