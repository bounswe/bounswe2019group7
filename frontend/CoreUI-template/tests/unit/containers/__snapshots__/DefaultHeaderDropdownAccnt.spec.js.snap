// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultHeaderDropdownAccnt.vue renders correctly 1`] = `
<appheaderdropdown-stub
  nocaret="true"
  right="true"
>
  <template>
    <img
      alt="admin@bootstrapmaster.com"
      class="img-avatar"
      src="img/avatars/6.jpg"
    />
  </template>
  \\
  
  <template>
    <b-dropdown-header-stub
      class="text-center"
      tag="div"
    >
      <strong>
        Account
      </strong>
    </b-dropdown-header-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-bell-o"
      />
       Updates
      
      <b-badge-stub
        event="click"
        routertag="a"
        tag="span"
        target="_self"
        variant="info"
      >
        42
      </b-badge-stub>
    </b-dropdown-item-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-envelope-o"
      />
       Messages
      
      <b-badge-stub
        event="click"
        routertag="a"
        tag="span"
        target="_self"
        variant="success"
      >
        42
      </b-badge-stub>
    </b-dropdown-item-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-tasks"
      />
       Tasks
      
      <b-badge-stub
        event="click"
        routertag="a"
        tag="span"
        target="_self"
        variant="danger"
      >
        42
      </b-badge-stub>
    </b-dropdown-item-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-comments"
      />
       Comments
      
      <b-badge-stub
        event="click"
        routertag="a"
        tag="span"
        target="_self"
        variant="warning"
      >
        42
      </b-badge-stub>
    </b-dropdown-item-stub>
     
    <b-dropdown-header-stub
      class="text-center"
      tag="div"
    >
      <strong>
        Settings
      </strong>
    </b-dropdown-header-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-user"
      />
       Profile
    </b-dropdown-item-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-wrench"
      />
       Settings
    </b-dropdown-item-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-usd"
      />
       Payments
      
      <b-badge-stub
        event="click"
        routertag="a"
        tag="span"
        target="_self"
        variant="secondary"
      >
        42
      </b-badge-stub>
    </b-dropdown-item-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-file"
      />
       Projects
      
      <b-badge-stub
        event="click"
        routertag="a"
        tag="span"
        target="_self"
        variant="primary"
      >
        42
      </b-badge-stub>
    </b-dropdown-item-stub>
     
    <b-dropdown-divider-stub
      tag="hr"
    />
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-shield"
      />
       Lock Account
    </b-dropdown-item-stub>
     
    <b-dropdown-item-stub
      event="click"
      routertag="a"
      target="_self"
    >
      <i
        class="fa fa-lock"
      />
       Logout
    </b-dropdown-item-stub>
  </template>
</appheaderdropdown-stub>
`;
